libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -421,10 +429,16 @@ AbstractAssembly::LabelID CodeTransform::labelID(Scope::Label const& _label)
 	return m_context->labelIDs[&_label];
 }
 
-AbstractAssembly::LabelID CodeTransform::functionEntryID(Scope::Function const& _function)
+AbstractAssembly::LabelID CodeTransform::functionEntryID(string const& _name, Scope::Function const& _function)
 {
 	if (!m_context->functionEntryIDs.count(&_function))
-		m_context->functionEntryIDs[&_function] = m_assembly.newLabelId();
+	{
+		AbstractAssembly::LabelID id =
+			m_useNamedLabelsForFunctions ?
```

Comment:
Any reason not to support both in the same assembly?

Reply:
I don't think that would work. If you use named tags, then function names have to be globally unique.

The problem is also that the code transform does not know whether the label will be referenced externally or not, so it has to use named tags for everything.

Perhaps I should have explained: This feature is there to build a bridge between IULIA and non-IULIA assembly. We want to be able to jump into IULIA-code from the `Assembly`-class generated code. Since IULIA will be translated into Assembly, we can do that.